Improved New Documentation

emersonbottero opened the pull request:
## :bookmark_tabs: Summary

New documentation migrating the last docs to vitepress:

- move docs from mermaid/src docs to mermaid/docs
- mermaid docs are still tranformed and copied to docs in the root folder
- added pipeline to publish docs to github pages (usually I have to algo go to setting and then publish to  github using gh-pages branch)
- move package scripts related to docs from root to mermaid package
- delete old script that creates setup.md for a new one `docs:code` (made some improvements in the jsdoc of ts files used here)
- remove lint from new docs (we can enable again later but must be sure the rules are ok first)
-  added meet the team in the main documentation page 😁
- reorganize all docs to folders following the current menu convention (it can be changed later for better organization)
- removed old v-docs (temporary created to show case vitepress)

## :straight_ruler: Design Decisions

suggestions made in https://github.com/mermaid-js/mermaid/issues/3610 are not included 


Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added unit/e2e tests (if appropriate)
- [x] :bookmark: targeted `develop` branch

sidharthv96 commented:
@emersonbottero can you add details of what exactly the changes are in the PR description?

emersonbottero commented:
> @emersonbottero can you add details of what exactly the changes are in the PR description?

done!

sidharthv96 reviewed (changes requested):
@emersonbottero This is a wonderful set of changes. Really appreciate the effort you've put into this 🚀 ❤️ 


> move docs from mermaid/src docs to mermaid/docs

Let's keep them inside `src/docs` as they are the source for the docs. 

> mermaid docs are still tranformed and copied to docs in the root folder

Please add the `note`, `tip`, etc to the transforms.

> remove lint from new docs 

Can you share which rules are problematic?

sidharthv96 commented:
Padding needs to be consistent.

https://user-images.githubusercontent.com/10703445/196239677-5c27332a-d213-4786-824e-3549a686f15c.mov

emersonbottero commented on packages/mermaid/docs/.vitepress/config.ts:
```diff
@@ -0,0 +1,162 @@
+    },
+    {
+      text: '💻 Live Editor',
+      link: '/edit',
```
-Let's keep them inside src/docs as they are the source for the docs. 
Done!
-Please add the note, tip, etc to the transforms
I'm already transforming.. maybe an example of what is missing?
-Use https://mermaid.live/?
-Can you share which rules are problematic?
can we try to add those later and see the errors?

emersonbottero commented on docs/config/usage.md:
````diff
@@ -357,9 +357,9 @@ approach are:
 ```
 
-```warning
+::: warning
````
actually vitepress need those to be ::: by default. 
did you meant to turn ::: in ``` in the transform step?
then it would be better to transform in 
```
> **Note**
> This is a note

> **Warning**
> This is a warning
```

that renders as:
> **Note**
> This is a note

> **Warning**
> This is a warning

emersonbottero replied:
done if it is ok with you

sidharthv96 replied:
> actually vitepress need those to be ::: by default.

This is the flow I had in mind. Is this not how it's done?

```mermaid
flowchart
src["/packages/mermaid/src/docs"] --> |Transform| docs["/docs"] --> |Vitepress| gh[GitHub pages]
docs --> g[Github Repo]
``` 

Is this how it's being done now?
```mermaid
flowchart
src["/packages/mermaid/src/docs"] --> |Transform| docs["/docs"] 
src--> |Vitepress| gh[GitHub pages]
docs --> g[Github Repo]
```

I would prefer the source docs to be proper markdown and not have any vitepress specific things.

emersonbottero replied:
Is the second one.. 

```warning doe not meat "warning" in github
if you remove the warning it is the same.. 
to do what you want we sould use 
```
> **Warning**
> Any text of warning block
```
we can try to do that later and change both transform but it is harder then the current version.

emersonbottero commented on docs/index.md:
```diff
@@ -0,0 +1,135 @@
+
+<div class="vp-doc" >
+  <h2 id="meet-the-team"> Meet The Team </h2>
+  <VPTeamMembers size="small" :members="members" />
```
I plan to add those later.. 
My intention was to to the same as in https://vitest.dev/

emersonbottero commented on packages/mermaid/docs/.vitepress/config.ts:
```diff
@@ -0,0 +1,162 @@
+
+    sidebar: {
+      '/': sidebarAll(),
+      // "/intro/": sidebarIntro(),
+      // "/syntax/": sidebarSyntax(),
+      // "/config": sidebarConfig(),
+      // "/misc/": sidebarMisc(),
+      // "/community/": sidebarCommunity(),
```
in case you want to change the side menu based on the path.. 
I thought it would be nice to leave it as a reference..

emersonbottero commented on packages/mermaid/docs/syntax/flowchart.md:
````diff
@@ -442,7 +449,7 @@ Examples of tooltip usage below:
 ```html
 <script>
-  const callback = function () {
+  var callback = function () {
````
done

emersonbottero commented on packages/mermaid/docs/.vitepress/config.ts:
```diff
@@ -0,0 +1,162 @@
+  description: 'Create diagrams and visualizations using text and code.',
+  base: '/mermaid-docs/',
+  markdown: MermaidMarkdown,
+  ignoreDeadLinks: true, //TODO: try to fixe those in autogenerated docs
```
Fixed.
forgot to remove the TODO..

emersonbottero commented on packages/mermaid/docs/.vitepress/mermaid-markdown-all.ts:
```diff
@@ -0,0 +1,32 @@
+    MermaidMarkdown(md);
+    const fence = md.renderer.rules.fence.bind(md.renderer.rules);
+
+    const highlighter = await shiki.getHighlighter({ theme: 'material-palenight' });
```
yes.

emersonbottero commented on docs/intro/index.md:
```diff
@@ -359,7 +347,7 @@ The above command generates files into the `dist` folder and publishes them to \
 Mermaid is a growing community and is always accepting new contributors. There's a lot of different ways to help out and we're always looking for extra hands! Look at [this issue](https://github.com/mermaid-js/mermaid/issues/866) if you want to know where to start helping out.
 
-Detailed information about how to contribute can be found in the [contribution guide](CONTRIBUTING.md)
+Detailed information about how to contribute can be found in the [contribution guide](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
```
when I try it breaks because after building there is not so much parent levels.

sidharthv96 replied:
Contribution is not part of the docs. No problem.

emersonbottero commented on packages/mermaid/docs/syntax/gantt.md:
```diff
@@ -275,7 +275,7 @@ todayMarker off
 
 This is done by defining the `ganttConfig` part of the configuration object.
-How to use the CLI is described in the [mermaidCLI](../config/mermaidCLI) page.
+How to use the CLI is described in the [mermaidCLI](../config/mermaidCLI.html) page.
```
the standard is no extension. 
but can be .d, .html or nothing at all

emersonbottero commented on packages/mermaid/package.json:
```diff
@@ -72,7 +78,10 @@
     "moment-mini": "^2.24.0",
     "non-layered-tidy-tree-layout": "^2.0.2",
-    "stylis": "^4.1.2"
+    "shiki": "^0.11.1",
+    "stylis": "^4.1.2",
+    "typedoc": "^0.23.16",
+    "typedoc-plugin-markdown": "^3.13.6"
```
done

emersonbottero commented:
> Padding needs to be consistent.
> 
>  Screen.Recording.2022-10-17.at.10.35.39.PM.mov 
> Copy code is a very useful feature. Can we also add an edit button, so the diagram opens in live editor? Can create a separate issue for this if it's complicated.
> 
> <img alt="image" width="280" src="https://user-images.githubusercontent.com/10703445/196240379-5dd69e3a-dcc0-41bf-aad5-6c51ee56bed3.png">
> 
> Overlap issue in gantt.
> 
> Can you add [#3610 (comment)](https://github.com/mermaid-js/mermaid/issues/3610#issuecomment-1272531180) logo ?

I removed some css I added to make the reading container larger.. I removed those .. 

Can you confirm the logo on the main page is ok as in the images I posted in https://github.com/mermaid-js/mermaid/issues/3610#issuecomment-1272531180?
I can remove the background color and left the default square logo.

for the favicon it will be the default square icon as it is.

emersonbottero commented:
please see the responses @sidharthv96

sidharthv96 commented on docs/community/newDiagram.md:
````diff
@@ -17,20 +21,17 @@ For instance:
-statement
-	: 'participant' actor  { $$='actor'; }
-	| signal               { $$='signal'; }
-	| note_statement       { $$='note';  }
-	| 'title' message      { yy.setTitle($2);  }
-	;
-```
+    statement
+    	: 'participant' actor  { $$='actor'; }
+    	| signal               { $$='signal'; }
+    	| note_statement       { $$='note';  }
+    	| 'title' message      { yy.setTitle($2);  }
+    	;
````
jison codeblock removed?

emersonbottero replied:
We dont have a highlight for now. and it brakes the doc build.

sidharthv96 replied:
There should be some option to ignore jison. 
Even if it's not possible, we should keep jison in source docs and strip just the jison language tag in transformed docs, keeping the code fences.

sidharthv96 commented on docs/community/newDiagram.md:
````diff
@@ -117,12 +118,12 @@ There are a few features that are common between the different types of diagrams
 Here is example handling from flowcharts:
 Jison:
 
-```jison
+```
````
jison?

sidharthv96 commented on docs/community/newDiagram.md:
````diff
@@ -179,27 +180,25 @@ The syntax for adding title and description looks like this:
-    : acc_title acc_title_value  { $$=$2.trim();yy.setTitle($$); }
-    | acc_descr acc_descr_value  { $$=$2.trim();yy.setAccDescription($$); }
-    | acc_descr_multiline_value { $$=$1.trim();yy.setAccDescription($$); }
-```
+    * lexical grammar */
````
Might be a bug in transformation?

sidharthv96 commented on docs/config/8.6.0_docs.md:
````diff
@@ -220,8 +220,8 @@ Ensures options parameter does not attempt to override siteConfig secure keys.
-```
+> **Note**\
+> default: current siteConfig (optional, default `getSiteConfig()`)
+> s
````
s?

sidharthv96 commented on docs/config/setup/modules/mermaidAPI.md:
```diff
@@ -0,0 +1,63 @@
+
+### <a id="mermaidapi" name="mermaidapi"></a> mermaidAPI
+
+• `Const` **mermaidAPI**: `Readonly`<{ `defaultConfig`: `MermaidConfig` = configApi.defaultConfig; `getConfig`: () => `MermaidConfig` = configApi.getConfig; `getSiteConfig`: () => `MermaidConfig` = configApi.getSiteConfig; `globalReset`: () => `void` ; `initialize`: (`options`: `MermaidConfig`) => `Promise`<`void`> ; `parse`: (`text`: `string`, `parseError?`: `Function`) => `boolean` ; `parseDire ...
```
Can this be formatted in any way?

sidharthv96 commented on docs/config/theming.md:
````diff
@@ -371,42 +374,6 @@ In the following examples, the directive `init` is used, with the `theme` being
           end
 ```
 
-```mermaid-example
-%%{init: {'securityLevel': 'loose', 'theme':'base'}}%%
-        graph TD
-          A[Christmas] -->|Get money| B(Go shopping)
-          B --> C{Let me think}
-          B --> G[/Another/]
-          C ==>|One| D[Laptop]
-          C -->|Two| E[iPhone]
-          C -->|Three| F[fa:fa-car Car]
-          subgraph section
-            C
-            D
-            E
-            F
-            G
-          end
-```
-
-```mermaid
-%%{init: {'securityLevel': 'loose', 'theme':'base'}}%%
-        graph TD
-          A[Christmas] -->|Get money| B(Go shopping)
-          B --> C{Let me think}
-          B --> G[/Another/]
-          C ==>|One| D[Laptop]
-          C -->|Two| E[iPhone]
-          C -->|Three| F[fa:fa-car Car]
-          subgraph section
-            C
````
Why was this removed?

sidharthv96 commented on docs/public/manifest.json:
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "App",
+  "icons": [
+    {
+      "src": "/android-icon-36x36.png",
+      "sizes": "36x36",
+      "type": "image/png",
+      "density": "0.75"
+    },
+    {
+      "src": "/android-icon-48x48.png",
+      "sizes": "48x48",
+      "type": "image/png",
+      "density": "1.0"
+    },
+    {
+      "src": "/android-icon-72x72.png",
+      "sizes": "72x72",
+      "type": "image/png",
+      "density": "1.5"
+    },
```
https://medium.com/swlh/are-you-using-svg-favicons-yet-a-guide-for-modern-browsers-836a6aace3df

We don't need all these icons.

sidharthv96 replied:
@emersonbottero please remove all the unnescessary icons.

sidharthv96 commented on packages/mermaid/package.json:
```diff
@@ -71,8 +77,7 @@
     "lodash": "^4.17.21",
     "moment-mini": "^2.24.0",
-    "non-layered-tidy-tree-layout": "^2.0.2",
-    "stylis": "^4.1.2"
```
stylis is an actual dependency of mermaid.

sidharthv96 commented on packages/mermaid/src/docs.mts:
```diff
@@ -131,6 +133,41 @@ const copyTransformedContents = (filename: string, doCopy = false, transformedCo
+    case 'note':
+      regex = /::: note\s?\n[\s\S]+?:::/gm;
+      break;
+    case 'tip':
+      regex = /::: tip\s?\n[\s\S]+?:::/gm;
+      text = 'Note';
+      break;
```
is text for tip `Note`?

emersonbottero replied:
markdown don't have the equivalente for tip
this is only used to created the docs in the root to be read inside github

sidharthv96 commented on packages/mermaid/src/docs/config/usage.md:
````diff
@@ -183,9 +183,9 @@ Or with no config object, and a jQuery selection:
 
-```warning
+::: warning
 This type of integration is deprecated. Instead the preferred way of handling more complex integration is to use the mermaidAPI instead.
-```
+:::
````
You should keep code block in the source as that's valid markdown. `:::` is not. 
Then, during transformation, you can check for code blocks with `['warning','tip','note'].includes(language)` and do the conversion instead of using regex.

emersonbottero replied:
it just fences the content in the transformed docs it shows with an Icon and i the webpage it has vitepress styles applied to it.

sidharthv96 commented on packages/mermaid/src/docs/syntax/flowchart.md:
```diff
@@ -495,10 +502,10 @@ Beginner's tip—a full example using interactive links in a html context:
   </pre>
 
   <script>
-    const callback = function () {
+    var callback = function () {
       alert('A callback was triggered');
     };
-    const config = {
+    var config = {
       startOnLoad: true,
```
These seem to still be var.

emersonbottero replied:
probably changed in the wrong place.
 done

sidharthv96 commented on packages/mermaid/src/docs/syntax/flowchart.md:
```diff
@@ -587,7 +594,7 @@ A shorter form of adding a class is to attach the classname to the node using th
 flowchart LR
     A:::someclass --> B
-    classDef someclass fill:#f96
+    classDef someclass fill:#f96;
```
semicolon.

emersonbottero replied:
done

sidharthv96 commented on packages/mermaid/src/docs/vite.config.ts:
```diff
@@ -0,0 +1,46 @@
+import { defineConfig, searchForWorkspaceRoot } from 'vite';
```
This config is duplicated.
You can add the one in /docs to gitignore as it's only used to generate vitepress and need not be checked in afterwards.

emersonbottero replied:
I cant remove it.. 
it is necessary to build the docs and is used in the ci/cd to deploy the page.

sidharthv96 commented:
> it is possible to make the links but I would have to understand how to generate the live editor link


You can refer [this](https://github.com/mermaid-js/mermaid-live-editor/blob/019542d1e818c3f524cd0c1c7ae9fb31a0e5c00f/src/lib/util/serde.ts#L38-L45) and use the base64 encoder. 
But let's do it after this one is merged.

sidharthv96 commented:
<img width="773" alt="image" src="https://user-images.githubusercontent.com/10703445/196359114-98350bba-c2a3-42de-91b5-fd6232dc4efd.png">
<img width="716" alt="image" src="https://user-images.githubusercontent.com/10703445/196359188-1e98441d-7826-488d-b8f6-47e0f0ebdaa1.png">

sidharthv96 commented:
How can we make sure that we have all the recent changes in docs?

emersonbottero commented on .gitignore:
```diff
@@ -32,3 +32,6 @@ cypress/snapshots/
 tsconfig.tsbuildinfo
+
+#docs generate from code
+packages/mermaid/src/docs/config/setup
```
I think I misunderstood the concept.
I was considering that:
1 - the documentation to be seen in the github should reside in /docs 
2 - the one to be seen in the gh pages would be in package/mermaid/src/docs 

that's why I keep the ::: warning in the src/docs and the > **Warning** in the /docs.
it is the same principal of mermaid-example in the src/docs and mermaid-example + mermaid in the /docs

with that in mind this folder is created before docs:build and before docs:bundle

is that correct?

emersonbottero commented on .lintstagedrc.json:
```diff
@@ -1,6 +1,4 @@
 {
-  "packages/mermaid/src/docs/**": ["pnpm run docs:build --git"],
-  "packages/mermaid/src/docs.mts": ["pnpm run docs:build --git"],
```
has been moved to the mermaid package, I can keep in both

emersonbottero commented on docs/.vitepress/config.ts:
```diff
@@ -1,17 +1,19 @@
   themeConfig: {
     nav: nav(),
+    editLink: {
+      pattern: 'https://github.com/mermaid-js/mermaid/edit/develop/docs/:path',
```
you are correct

emersonbottero commented:
> <img alt="image" width="773" src="https://user-images.githubusercontent.com/10703445/196359114-98350bba-c2a3-42de-91b5-fd6232dc4efd.png">
> 
> <img alt="image" width="716" src="https://user-images.githubusercontent.com/10703445/196359188-1e98441d-7826-488d-b8f6-47e0f0ebdaa1.png">

didn't understood

emersonbottero commented:
> How can we make sure that we have all the recent changes in docs?

I get the last one and changed.. 
but I guess if something is missing we can add it later.. 
there are too many alterations at once in the new docs

emersonbottero commented:
> Padding needs to be consistent.
> 
>  Screen.Recording.2022-10-17.at.10.35.39.PM.mov 
> Copy code is a very useful feature. Can we also add an edit button, so the diagram opens in live editor? Can create a separate issue for this if it's complicated.
> 
> <img alt="image" width="280" src="https://user-images.githubusercontent.com/10703445/196240379-5dd69e3a-dcc0-41bf-aad5-6c51ee56bed3.png">
> 
> Overlap issue in gantt.
> 
> Can you add [#3610 (comment)](https://github.com/mermaid-js/mermaid/issues/3610#issuecomment-1272531180) logo ?

you where correct.. I fixed the width error (probably will also be fixed in vitepress soon)
I added the logo the same way as where in the images I sent in the mentioned link

sidharthv96 commented:
> didn't understood

Just small issue. 

The lines with > should be in new lines.

<img width="101" alt="image" src="https://user-images.githubusercontent.com/10703445/196733548-07c34545-fb82-420f-802f-7a1c6ec7c08d.png">

Extra `**`
<img width="49" alt="image" src="https://user-images.githubusercontent.com/10703445/196733632-fa6ec9e7-ac04-4208-ba49-ea6b09c1df97.png">

emersonbottero commented on CONTRIBUTING.md:
```diff
@@ -152,7 +152,7 @@ The source files for documentation are in `/packages/mermaid/src/docs` and are w
 #### Adding to or changing the documentation organization
 
-If you want to add a new section or change the organization (structure), then you need to make sure to **change the side navigation** in `src/docs/_sidebar.md`.
+If you want to add a new section or change the organization (structure), then you need to make sure to **change the side navigation** in `docs/_sidebar.md`.
```
actually updated to vitepress

emersonbottero commented:
@sidharthv96  I think now all is done!
hopefully! 😁

sidharthv96 commented:
I hope these are the last set of changes. 

Additional comments from previous reviews.

- https://github.com/mermaid-js/mermaid/pull/3678#discussion_r997699715


Thanks for putting in all the effort and making all the suggested changes!! 🚀

emersonbottero commented on docs/config/setup/modules/mermaidAPI.md:
```diff
@@ -0,0 +1,63 @@
+
+#### Defined in
+
+[mermaidAPI.ts:72](https://github.com/emersonbottero/mermaid/blob/455839c0/packages/mermaid/src/mermaidAPI.ts#L72)
```
I disabled the link creation since I couldn't fix it

emersonbottero commented on docs/config/setup/modules/defaultConfig.md:
````diff
@@ -0,0 +1,56 @@
+
+```html
+<script>
+  var config = {
````
comes from defautlConfig.ts line 21

sidharthv96 commented:
@emersonbottero please check if I've broken anything.

sidharthv96 commented:
New flow.

```mermaid
flowchart
src["/packages/mermaid/src/docs"] --> |docs:build| docs["/docs"] 
src--> |docs:build:vitepress| vp["/packages/mermaid/src/vitepress"]
vp --> |docs:bundle| gh[GitHub pages]
docs --> g[Github Repo]
```

aloisklink reviewed (commented):
Wow, the new website looks amazing!

Thanks for putting in all of this work!

There's a couple of issues/potential improvements that I found.

Some of these might be better to fix in a future PR though, since otherwise I'm sure we'd be battling lots of merge conflicts by not merging this PR.

### Issues

The sidebar link to git graphs goes to a 404 page:

The following line needs to be `gitgraph`, not `gitGraph`:

https://github.com/mermaid-js/mermaid/blob/fed32607dea1cbe3db24fe6b67ecdc7ceed3c173/packages/mermaid/src/docs/.vitepress/config.ts#L107

(if you're using a Windows computer, they're identical, but Linux/MacOS file paths are case sensitive)

---

The sidebar link to CHANGELOG goes to a 404 page

https://github.com/mermaid-js/mermaid/blob/fed32607dea1cbe3db24fe6b67ecdc7ceed3c173/packages/mermaid/src/docs/.vitepress/config.ts#L159

It's probably find to delete, since there's already a link to the CHANGELOG in the header

![image](https://user-images.githubusercontent.com/19716675/198896731-6f9f9db2-944b-41ff-bf3d-4c2d8dae35dc.png)

---

The `Mermaid API Configuration` sidebar link is broken:

https://github.com/mermaid-js/mermaid/blob/fed32607dea1cbe3db24fe6b67ecdc7ceed3c173/packages/mermaid/src/docs/.vitepress/config.ts#L123

It should point to `config/setup/README` (e.g. the `typedoc` output folder)

---

The typedoc output is kinda bad right now. It's missing pretty much all the docs that are at https://mermaid-js.github.io/mermaid/#/Setup?id=mermaidapi

My guess is that we have to convert all of our `tsdoc` to use things like [`@defaultValue ....`](https://tsdoc.org/pages/tags/defaultvalue/), but that will probably be a bunch of work to do.

As a short-term measure, can we add some links to the source-code?

E.g., when visiting /mermaid-js/config/setup/modules/config.html (and other files) it says something like:

> **Warning**
>
> We are still in the process of automatically generating these docs from the source-code. Please see the source file at <https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/config.ts> for full documentation.

It might also be worth trying out [@microsoft/api-documenter](https://www.npmjs.com/package/@microsoft/api-documenter) and see if that's better than using typedoc.

Something we'd also want is to point the `Edit this page on GitHub` link to the source-files, but that might also be a lot of work!

emersonbottero commented on .github/gh-pages-publish.yml:
```diff
@@ -0,0 +1,30 @@
+          cd ./packages/mermaid
+          pnpm docs:build:vitepress
+
+      - name: Deploy 🚀📕
+        uses: JamesIves/github-pages-deploy-action@v4.3.3
+        with:
+          branch: gh-pages
+          folder: packages/mermaid/src/vitepress/.vitepress/dist
```
deleted

emersonbottero replied:
I also fixed the ts file so defaults values now also appears in the docs (In case this was the only problem)
It should be before a method to be used.

aloisklink replied:
> fixed the topis above 

Great! Thanks for the fixes!

> deleted the gh deploy yml file

I'm okay for having a `gh-deploy.yml` file in this PR if it saves maintainer's time. However, for a future PR, we should change the `JamesIves/github-pages-deploy-action` step to the official GitHub `actions/deploy-pages`.

> I also fixed the ts file so defaults values now also appears in the docs (In case this was the only problem)
> It should be before a method to be used.

It's better, but it's still missing quite a bit compared to the old https://mermaid-js.github.io/mermaid/#/Setup page.

E.g. these docs about the `logLevel` configuration option is nowhere on the new site: 
![image](https://user-images.githubusercontent.com/19716675/198914050-6a52e145-7e00-4e5a-bfd5-08f25a814a9b.png)

I think the reason why these docs are not getting exported is because the TSDoc comments are on a value, not on a type.

We might need to rewrite the TSDoc comments (or convert `config.ts` and `defaultConfig.ts` to use TypeScript classes instead of being a plain object) to get this exported properly to markdown. But this seems like a lot of work, so we should leave it to a future PR.

Unless somebody can think of a quick fix, for this PR, I think we should be okay with just adding a link to <https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/defaultConfig.ts>, and tell people that the full docs are there.

emersonbottero commented on .github/gh-pages-publish.yml:
```diff
@@ -0,0 +1,30 @@
+      - name: Install and Build 🔧
+        run: |
+          cd ./packages/mermaid
+          pnpm docs:build:vitepress
```
deleted the file

emersonbottero commented on .github/workflows/lint.yml:
```diff
@@ -39,4 +39,4 @@ jobs:
 
       - name: Verify Docs
-        run: pnpm run docs:verify
+        run: cd packages/mermaid && pnpm docs:verify
```
changed

emersonbottero commented on CONTRIBUTING.md:
```diff
@@ -67,31 +67,23 @@ flowchart LR
-$ npm i docsify-cli -g
+$ cd mermaid
+$ pnpm i
+$ pnpm docs:dev
```
fixed. 
this one is outside of the documentation build with vitepress it links to github.. 
but shikki, the highlighter does not not the console language

emersonbottero commented on CONTRIBUTING.md:
````diff
@@ -67,31 +67,23 @@ flowchart LR
 ```sh
-$ yarn global add docsify-cli
+$ pnpm docs:build:vitepress
+$ pnpm docs:serve
````
done

emersonbottero commented on packages/mermaid/package.json:
```diff
@@ -27,21 +27,25 @@
-    "todo-postbuild": "documentation build src/mermaidAPI.ts src/config.ts src/defaultConfig.ts --shallow -f md --markdown-toc false > src/docs/Setup.md && prettier --write src/docs/Setup.md",
-    "release": "yarn build",
-    "lint": "eslint --cache --ignore-path .gitignore . && yarn lint:jison && prettier --check .",
+    "docs:code": "typedoc --plugin typedoc-plugin-markdown --readme none --githubPages false --hideBreadcrumbs --gitRemote disable --hideInPageTOC --namedAnchors --out src/docs/config/setup --entryPointStrategy expand src/defaultConfig.ts src/config.ts src/mermaidAPI.ts && prettier --write ./src/docs/config/setup",
```
done

emersonbottero commented on packages/mermaid/src/docs.mts:
```diff
@@ -42,22 +42,24 @@ import { remark } from 'remark';
-
-const verifyOnly: boolean = process.argv.includes('--verify');
-const git: boolean = process.argv.includes('--git');
+const WARN_DOCSDIR_DOESNT_MATCH = `Changed files were transformed in ${SOURCE_DOCS_DIR} but do not match the files in ${FINAL_DOCS_DIR}. Please run pnpm docs:build after making changes to ${SOURCE_DOCS_DIR} to update the ${FINAL_DOCS_DIR} directory with the transformed files.`;
```
I'll leave this one to @sidharthv96

emersonbottero commented on packages/mermaid/src/docs.mts:
```diff
@@ -72,11 +74,13 @@ let filesWereTransformed = false;
+  const filePathFromRoot = posix.join('/packages/mermaid', file);
+  const sourcePathRelativeToGenerated = posix.join(relativePath, filePathFromRoot);
+  return `
+> **Warning**
+> ## THIS IS AN AUTOGENERATED FILE. DO NOT EDIT. 
+> ## Please edit the corresponding file in [${filePathFromRoot}](${sourcePathRelativeToGenerated}).`;
```
I also think so.. note also works 🙂

emersonbottero commented on packages/mermaid/src/docs/config/setup/README.md:
```diff
@@ -0,0 +1,7 @@
+# mermaid
+
+## Modules
+
+- [config](modules/config.md)
+- [defaultConfig](modules/defaultConfig.md)
+- [mermaidAPI](modules/mermaidAPI.md)
```
I'll leave this one to @sidharthv96

sidharthv96 replied:
Yup, ignored the `src/docs`, kept in `docs/`.
I think I removed the ignore @emersonbottero had already added.

emersonbottero commented on packages/mermaid/package.json:
```diff
@@ -27,21 +27,25 @@
+    "docs:pre:vitepress": "rimraf src/vitepress && pnpm docs:code && ts-node-esm src/docs.mts --vitepress",
+    "docs:build:vitepress": "pnpm docs:pre:vitepress && vitepress build src/vitepress",
+    "docs:dev": "pnpm docs:pre:vitepress && vitepress dev src/vitepress",
+    "docs:serve": "pnpm docs:pre:vitepress && vitepress serve src/vitepress",
```
I'll leave this one to @sidharthv96

sidharthv96 commented:
> @aloisklink Something we'd also want is to point the Edit this page on GitHub link to the source-files, but that might also be a lot of work!

The current link points to `https://github.com/mermaid-js/mermaid/edit/develop/packages/mermaid/src/docs/intro/index.md` which should be the correct link once the PR is merged.

sidharthv96 commented:
@emersonbottero, seems like you removed the deploy action. How will it be pushed to github pages? 🤔

sidharthv96 commented:
Deployed to https://sidharthv96.github.io/mermaid

mmorel-35 commented:
![image](https://user-images.githubusercontent.com/6032561/199066693-ad857577-8097-4d92-93a9-4a9f0363ca87.png)

On flowchart

sidharthv96 commented:
Homepage
<img width="598" alt="image" src="https://user-images.githubusercontent.com/10703445/199067275-35c25879-4786-40bc-990b-611f25d477a6.png">

---

Class Diagrams
<img width="576" alt="image" src="https://user-images.githubusercontent.com/10703445/199067398-1ab915ed-3621-4541-8710-d5867bd9da1b.png">

mmorel-35 commented:
Here are all the errors I could see from the chrome console while browsing through https://sidharthv96.github.io/mermaid
